### DISCUSSION TITLE
Accessing loop variables

### DISCUSSION THREAD

**Discussion by DoreenMulder** [mitsuba-renderer/mitsuba3#525] [Q&A]
Hi,

Over the last few days I've tried to implement a Mie scattering phase function. The size parameter I need to calculate is wavelength-dependent, so I'm trying to get these wavelengths from the MediumInteraction that is passed to me in the eval(ctx, mi, wo, active) method. 
I'm struggling with accessing the wavelengths this way, because DrJit says it's a placeholder variable and cannot be scheduled for evaluation. Can anyone point me in the right direction here?

Thanks!

**Comment by njroussel** [comment]
Hi @DoreenMulder 

Could you share a snippet of your code? 
The Dr.Jit message indicates that you're performing some sort of an operation which must be evaluated immediately in order for the program to continue. For example, even a simple `print(var)` in your code will trigger an evaluation.

**Reply by njroussel** [reply]
If this sounds confusing, take a look at [Dr.Jit's documentation](https://drjit.readthedocs.io/en/latest/intro.html). It's important to keep in mind that when writing code with Dr.Jit types, you are now working with a Just-In-Time compiler.

**Reply by DoreenMulder** [reply]
Thanks for the reply @njroussel ! 

I've read the documentation but I think I'm misunderstanding something that I can't seem to figure out by myself.

I'm trying to use drjit.schedule on `mi.wavelengths` to set the wavelenghts up for later evaluation. 
For debugging purposes I tried to schedule only the first wavelength, but `dr.schedule(mi.wavelengths)` behaves the same. This is the line that throws the error in my `eval()` method:

`x = self.calc_size_param(self.r, dr.schedule(mi.wavelengths[0]))`

with `calc_size_param()` defined as follows:

```
def calc_size_param(self, r, wl):
        val = 2 * dr.pi * r / wl
        return val 
```

I'm using `dr.schedule()` on a few other variables which do seem to work, and I'm having a hard time finding out why wavelengths gives me an error. 

Thanks for your help!

**Reply by njroussel** [reply]
Ok, the `calc_size_param` operation seems fine. It should not trigger an undesired evaluation.

My best guess is that the `mi` variable (the `MediumInteraction`) is a state variable in your loop?
